content/mqtt/index.textile

```diff
@@ -0,0 +1,103 @@
+    - Tutorials#tutorials
+---
+
+The "Ably MQTT protocol adapter":https://www.ably.io/adapters is able to convert communication from "MQTT":http://mqtt.org/ to Ably's own protocol, allowing for seamless integration of any systems you may have. MQTT (MQ Telemetry Transport) is a publish/subscribe, lightweight messaging protocol designed for constrained devices and low-bandwidth networks. One of the major uses of MQTT is with IoT (Internet of Things), where these principles are key to having effective communication between various devices.
+
+h2(#use). How to use MQTT with our adapter
+
+To use the Ably MQTT protocol adapter, you'll need to ensure you correctly configure your MQTT client as follows:
+
+* Set the host to "mqtt.ably.io"
+* Set ssl / tls to true and the port to 8883. (If your mqtt client does not support ssl, you should instead use port 1883, but in this case we disallow api-key auth—see "SSL usage note":#ssl below)
+* Set the keep alive time to somewhere between 15 and 60 seconds. (60s will maximise battery life, 15s will maximise reponsiveness to network issues. It must not be any higher than 60s to avoid our load balancer terminating the TCP socket for inactivity)
+* If using an api key, set the username to the part of the api key before the colon, and the password to the part after the colon
+* If using a token, set the username to the token, and leave the password blank
+
+For example, in JavaScript, you'd need to specify the following:
+
+```[javascript]
+var options = { keepAlive: 15,
+  username: 'FIRST_HALF_OF_API_KEY',
+  password: 'SECOND_HALF_OF_API_KEY',
+  port: 8883
+};
+var client = mqtt.connect('mqtts:mqtt.ably.io', options);
+```
+
+This will connect using TLS through MQTT to Ably. If you wanted to not use SSL, you'd use port 1883 instead.
+
+h2(#decoding). Decoding MQTT communication
+
+The data sent to Ably will be binary encoded, due to MQTT being a binary protocol. This means that you'll need to interpret the message to get the original contents out. For example, if you tried to send a string with MQTT, with JavaScript you'd need to do the following:
+
+```[javascript]
+var decoder = new TextDecoder();
+var command = decoder.decode(message.data);
+```
+
+In the above example, @command@ will now contain the message in its original string form.
+
+h2(#use-cases). Recommended usage
+
+We recommend you use MQTT to interact with Ably when:
+* Using devices which do not have a native Ably client library, such as Arduino platforms, C/C++ applications, and so on
+* If you need to work with a very low bandwidth, or want to keep your network traffic to a minimum
+
+h2(#supported). Supported features
+
+MQTT is only able to make use of the basic Realtime pub/sub mechanics of the Ably system. For all other features, such as presence, history and push notifications, you'll need to make use of an Ably client library. We will soon be allowing for devices to be present in channels when connected through MQTT.
```

Comment:
I'd rework this para to note that this applies both ways round. Maybe something like:

"Our MQTT adapter only supports features supported by both the MQTT protocol and the Ably platform. So:
- It doesn’t support any MQTT features that aren’t normally supported by Ably, such as RETAIN messages or wildcard channel subscriptions
- It doesn't support any Ably features that aren't supported by the MQTT protocol, such as presence, history and push notifications (though presence may be supported soon). Of course, you can always use the "Ably REST API":https://www.ably.io/documentation/rest-api in conjunction with the MQTT adapter to access features available over REST."